mmv1/third_party/terraform/provider/provider_billing_project_test.go

```diff
@@ -0,0 +1,326 @@
+		"random_suffix":         randomString,
+	}
+
+	contextUserProjectOverrideTrue := map[string]interface{}{
+		"org_id":                envvar.GetTestOrgFromEnv(t),
+		"billing_account":       envvar.GetTestBillingAccountFromEnv(t),
+		"user_project_override": "true", // Used in combo with billing_account
+		"random_suffix":         randomString,
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		// No PreCheck for checking ENVs
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				// Setup resources
+				// Neither user_project_override nor billing_project value used here
+				Config: testAccSdkProvider_billing_project_useBillingProject_setup(contextUserProjectOverrideFalse),
+			},
+			{
+				// With user_project_override=true the PubSub topic CANNOT be provisioned because quota is consumed
+				// from the newly provisioned project, and that project does not have the PubSub API enabled.
+				// The billing_project is used, leading to the error occurring, because user_project_override=true
+				Config:      testAccSdkProvider_billing_project_useBillingProject_scenario(contextUserProjectOverrideTrue),
+				ExpectError: regexp.MustCompile(fmt.Sprintf("Error 403: Cloud Pub/Sub API has not been used in project tf-test-%s", randomString)),
+			},
+			{
+				// With user_project_override=false the PubSub topic can be provisioned because quota is consumed
+				// from the project the Terraform identity is in, and that project has PubSub API enabled.
+				// The billing_project value isn't used, meaning the error doesn't happen, because user_project_override=false
+				Config: testAccSdkProvider_billing_project_useBillingProject_scenario(contextUserProjectOverrideFalse),
+			},
+		},
+	})
+}
+
+func testAccSdkProvider_billing_project_affectedByClientLibraryEnv(t *testing.T) {
+	randomString := acctest.RandString(t, 10)
+
+	context := map[string]interface{}{
+		"org_id":                envvar.GetTestOrgFromEnv(t),
+		"billing_account":       envvar.GetTestBillingAccountFromEnv(t),
+		"user_project_override": "true", // Used in combo with billing_account
+		"random_suffix":         randomString,
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		// No PreCheck for checking ENVs
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				// Setup resources
+				// Neither user_project_override nor billing_project value used here
+				Config: testAccSdkProvider_billing_project_useBillingProject_setupWithApiEnabled(context),
+			},
+			{
+				// This ENV interferes with setting the billing_project,
+				// so we get an error mentioning the value
+				PreConfig: func() {
+					t.Setenv("GOOGLE_CLOUD_QUOTA_PROJECT", "foobar")
+				},
+				Config:      testAccSdkProvider_billing_project_useBillingProject_scenarioWithApiEnabled(context),
+				ExpectError: regexp.MustCompile("Caller does not have required permission to use project foobar"),
+			},
```

Comment:
[This test failure](https://github.com/GoogleCloudPlatform/magic-modules/pull/11610#issuecomment-2386175893) is because this step isn't failing as expected